src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -330,11 +330,14 @@ export class NotebookEditor implements INotebookEditor {
         } catch (exc) {
             // If we get a kernel promise failure, then restarting timed out. Just shutdown and restart the entire server.
             // Note, this code might not be necessary, as such an error is thrown only when interrupting a kernel times out.
+            sendKernelTelemetryEvent(
+                this.document.uri,
+                Telemetry.NotebookRestart,
+                stopWatch.elapsedTime,
+                undefined,
+                exc
+            );
             if (exc instanceof JupyterKernelPromiseFailedError && kernel) {
-                sendKernelTelemetryEvent(this.document.uri, Telemetry.NotebookRestart, stopWatch.elapsedTime, {
-                    failed: true,
```

Comment:
This is not required anymore, just send the error & the telemetry code will determine the error.